packages/asc-ui/src/components/Menu/MenuFlyOut/MenuFlyOut.test.tsx

```diff
@@ -0,0 +1,152 @@
+  const useMenuContextMock = useMenuContext as jest.Mock
+  const useMenuFocusMock = useMenuFocus as jest.Mock
+
+  const children = [
+    <MenuItem key={1} href="/">
+      Space Bear 6
+    </MenuItem>,
+    <MenuItem key={2} href="/">
+      Space Bear 6 Plus
+    </MenuItem>,
+    <MenuItem key={3} href="/">
+      Space Bear 7
+    </MenuItem>,
+  ]
+
+  const mockContext = {
+    activeChild: 0,
+    setActiveChild: jest.fn(),
+    setActiveToggleChild: jest.fn(),
+    setOpenToggle: jest.fn(),
+    underFlyOutMenu: false,
+    hasToggle: false,
+  }
+
+  const onClickSimulate = (element: any = component) =>
+    element
+      .find(MenuItemLink)
+      .at(0)
+      .simulate('click', { preventDefault: () => {} })
+  const onKeyDownSimulate = (keyType: any) =>
+    component
+      .find(MenuItemLink)
+      .at(0)
+      .simulate('keydown', {
+        preventDefault: () => {},
+        key: KeyboardKeys[keyType],
+      })
+
+  beforeEach(() => {
+    useMenuContextMock.mockImplementation(() => mockContext)
+    useMenuFocusMock.mockImplementation(() => {})
+
+    component = shallow<{}>(<MenuFlyOut label="Fly Out">{children}</MenuFlyOut>)
+  })
+
+  it('should render', () => {
+    expect(component.exists()).toBeTruthy()
+  })
+
+  it('should be closed by default', () => {
+    menuListProps = component.find(MenuList).props()
+    expect(menuListProps['aria-hidden']).toBe(true)
+  })
+
+  it('should toggle the open state on click', () => {
+    onClickSimulate()
+
+    menuListProps = component.find(MenuList).props()
+    expect(menuListProps['aria-hidden']).toBe(false)
+  })
+
+  it('should not toggle the open state when already opened', () => {
+    onClickSimulate()
+    onClickSimulate()
+
+    menuListProps = component.find(MenuList).props()
+    expect(menuListProps['aria-hidden']).toBe(false)
+  })
+
+  it('should toggle the open state on enter', () => {
+    onKeyDownSimulate('Enter')
+
+    menuListProps = component.find(MenuList).props()
+    expect(menuListProps['aria-hidden']).toBe(false)
+  })
+
+  it('should toggle the open state on space', () => {
+    onKeyDownSimulate('Space')
+
+    menuListProps = component.find(MenuList).props()
+    expect(menuListProps['aria-hidden']).toBe(false)
+  })
+
+  it('should call onExpand', () => {
+    const onExpandMock = jest.fn()
+    useMenuContextMock.mockImplementationOnce(() => ({
+      ...mockContext,
+      onExpand: onExpandMock,
+    }))
+
+    container = mount<{}>(<MenuFlyOut label="Fly Out">{children}</MenuFlyOut>)
+
+    onClickSimulate(container)
+
+    expect(onExpandMock).toHaveBeenCalled()
+  })
+
+  describe('when used nested', () => {
+    const setActiveToggleChildMock = jest.fn()
+
+    beforeEach(() => {
+      useMenuContextMock.mockImplementation(() => ({
+        ...mockContext,
+        setActiveToggleChild: setActiveToggleChildMock,
+        hasToggle: true,
+      }))
+      useMenuFocusMock.mockImplementation(() => {})
+
+      component = shallow<{}>(
```

Comment:
Doesn't this already happen in the upper scope?

Reply:
The difference is the hasToggle field in the Context, this means the MenuFlyOut is used in a mobile menu